plugins/isecl/client/isecl-plugin.go

```diff
@@ -0,0 +1,493 @@
+	}
+	if c.ClientKey == "" {
+		c.ClientKey = defaultClientKey
+	}
+	if c.CaCert == "" {
+		c.CaCert = defaultCaCert
+	}
+	if c.KbsEndpoint == "" {
+		c.KbsEndpoint = defaultKbsEndpoint
+	}
+	if c.SqvsEndpoint == "" {
+		c.SqvsEndpoint = defaultSqvsEndpoint
+	}
+	if c.ConfigPath == "" {
+		if _, err := os.Stat(defaultConfigPath); err == nil {
+			c.ConfigPath = defaultConfigPath
+		}
+	}
+	if c.Timeout == 0 {
+		c.Timeout = defaultTimeout * time.Second
+	}
+}
+
+type quoteData struct {
+	// QuoteBlob is base64 encoded quote excluding it's length
+	QuoteBlob string `json:"quote"`
+	// UserData is base64 encoded of publickey+nonce
+	UserData string `json:"userData"`
+}
+
+type iSeclPlugin struct {
+	config  *Config
+	log     logr.Logger
+	client  httpclient.HttpClient
+	signers config.SignersConfig
+}
+
+var _ plugin.KeyManagerPlugin = &iSeclPlugin{}
+
+func NewISecLPlugin(config *Config) (plugin.KeyManagerPlugin, error) {
+	if config == nil {
+		// create new empty config
+		config = &Config{}
+	}
+	// set default values for not provided parameters
+	config.setDefaults()
+
+	km := &iSeclPlugin{
+		log:    klogr.New().WithName("isecl"),
+		config: config,
+	}
+	if err := km.initialize(); err != nil {
+		return nil, err
+	}
+	return km, nil
+}
+
+func (km *iSeclPlugin) initialize() error {
+	if km == nil {
+		return errors.New("kmStore is nil")
+	}
+	if km.config.ConfigPath != "" {
+		signers, err := config.ParseConfigFile(km.config.ConfigPath)
+		if err != nil {
+			return err
+		}
+		km.signers = signers
+	} else {
+		km.signers = make(config.SignersConfig)
+	}
+	// Create new http client to send requests to keys server
+	client, err := httpclient.NewHttpClient(km.config.CaCert, km.config.ClientCert, km.config.ClientKey, km.config.Timeout)
+	if err != nil {
+		return err
+	}
+	km.client = client
+
+	// Get the server API version
+	version, err := km.getApiVersion()
+	if err != nil {
+		return err
+	}
+	km.log.Info("server version response", "KBS Version", version)
+	// Check if api version meets criteria
+	//err = validateMinApiVersion(version)
+	//if err != nil {
+	//	return fmt.Errorf("server api version validation failed: %v", err)
+	//}
+
+	keys, err := km.fetchKeys()
+	if err != nil {
+		return fmt.Errorf("failed to get existing key info: %v", err)
+	}
+
+	for _, key := range keys {
+		if key.Label == "" {
+			continue
+		}
+		km.log.Info("Signer Key", "ID", key.KeyInformation.ID, "label", key.Label)
+		caInfo, ok := km.signers[key.Label]
+		if !ok {
+			caInfo = config.CAInfo{}
+		}
+		caInfo.KeyID = key.KeyInformation.ID.String()
+		km.signers[key.Label] = caInfo
+	}
+	return nil
+}
+
+// AttestQuote uses post request to attest the quote match the key for a given signer
+func (km *iSeclPlugin) AttestQuote(signerName string, encodedQuote []byte, publicKey []byte) (bool, error) {
+	if km.client == nil {
+		return false, errors.New("http client is not initialized")
+	}
+
+	/*token, err := km.fetchBearerToken()
```

Comment:
Just remove commented out parts of the code (here and thirty lines above this) if they are not needed.

Reply:
Done